### ISSUE TITLE
Evaluate use of `listbox` and keyboard navigation of customize columns tearsheet

### ISSUE THREAD

**Issue by elycheea** [carbon-design-system/ibm-products#4781]
Reported by @kristastarr in the Datagrid [customize columns review](https://github.ibm.com/CDAI-design/pal/issues/1576) from the accessibility audit.

Note that these were flagged as **potential violations** and **recommendations**.

> Confirm the `<ol> <listbox>` has keyboard access and focus indicator - **_When the find column search bar has focus, and you press tab, there is a tab stop that I think is the select all checkbox, but has no focus indicator so is not clear. You reach the checkbox after you press tab again which does have a correct focus indicator._**

Looks like the whole listbox has tab access but no focus indicator. This may be to support navigation in the modal via keyboard if the column list becomes long enough to require scrolling.

Also while the `<ol>` does use an `aria-label`, we can probably use `aria-labelledby` with the tearsheet title in this case.

- Verify the `<ol>` element with `"listbox"` role has keyboard access
- Check the keyboard focus indicator is visible when using CSS declaration for `border` or `outline`
- The input element does not have an associated visible label
- Component with `listbox` role has more than one tabbable element
- For the <li>s - the element with the role “option” contains implicit roles “checkbox” which are ignored by browsers
- The element with role "listbox" owns the child element with the role "checkbox" that is not one of the allowed role(s): "option"
- The element with role "listbox" owns the child element with the role "status" that is not one of the allowed role(s): "option"

The customize columns tearsheet contains both arrow keys and tabbable checkboxes for selecting which columns to display. It seems this may not be appropriate for the role `listbox`.

From Equal Access:

> ### Why is this important?
> Unlike native HTML form elements, browsers do not provide keyboard support for graphical user interface (GUI) components that are made accessible with ARIA. Authors need to implement keyboard access using keyboard handlers, such as `onkeydown` or `onkeypress`.
>
> ### Element location
> ```html
> <ol
>    tabindex="0"
>    aria-describedby="c4p--datagrid__customize-columns--instructions"
>    aria-label="Customize columns"
>    role="listbox"
>    class="c4p--datagrid__customize-columns-column-list--focus">
>```
>
> All focus indicators should be visible and distinguishable. When navigating a Web page with a keyboard, users need to know which element has the keyboard focus. A visual focus indicator might be a vertical bar (a typical cursor) or a visible border surrounding a user interface control.

**Comment by elycheea** [comment]
At a quick glance, we may need to restructure the HTML a bit so that the `listbox` only contains the list items themselves and not also the divs and spans. Use of checkboxes within the listbox options also seems invalid.

**Comment by ljcarot** [comment]
We've been investigating [TanStack ](https://tanstack.com/table/latest)[Table](https://tanstack.com/table/latest), a third-party, open-source offering, which provides extensive data table capabilities surpassing what our Carbon Datagrid offers. It provides much more flexibility and customization. TanStack Table is headless which means it can easily be added alongside Datagrid component in your product or application. The benefits of more flexibility for product teams and less maintenance for Carbon makes it a win win. Lastly, it is available in multiple frameworks including [React](https://github.com/carbon-design-system/tanstack-carbon/tree/main/react#readme) and [Web Component](https://github.com/carbon-design-system/tanstack-carbon/tree/main/web-components#readme) so it provides an option to non-React product teams.

For these reasons, we have decided to transition from building our own custom table component to using an example-based approach with TanStack Table. Datagrid will still exist in our library for existing teams but we are announcing the deprecation* of the Datagrid component in v2.54.0 release so teams can begin to work through the transition. Details about how to use both Datagrid and TanStack together can be found [here](https://github.com/carbon-design-system/tanstack-carbon/tree/main). 

*Deprecation means that no new features will be added however sev 1 and sev 2 bugs will be supported.
